mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -102,33 +151,80 @@ class Completions(
   inline private def undoBacktick(label: String): String =
     label.stripPrefix("`").stripSuffix("`")
 
-  private def findSuffix(methodSymbol: Symbol): Option[String] =
-    if shouldAddSnippet && methodSymbol.is(Flags.Method) then
-      lazy val extensionParam = methodSymbol.extensionParam
+  private def getParams(symbol: Symbol) =
+    lazy val extensionParam = symbol.extensionParam
+    if symbol.is(Flags.Extension) then
+      symbol.paramSymss.filter(
+        !_.contains(extensionParam)
+      )
+    else symbol.paramSymss
+
+  private def findSuffix(symbol: Symbol): Option[String] =
+
+    val bracketSuffix =
+      if shouldAddSnippet && noSquareBracketExists
+        && (isTypePosition || isNewPosition || isInstantiationOrMethodCallPos)
+        && (symbol.info.typeParams.nonEmpty
+          || (symbol.isAllOf(
+            Flags.JavaModule
+          ) && symbol.companionClass.typeParams.nonEmpty))
+      then "[$0]"
+      else ""
+
+    val bracesSuffix =
+      if shouldAddSnippet && symbol.is(
+          Flags.Method
+        ) && !isTypePosition
+      then
+        val paramss = getParams(symbol)
+        paramss match
+          case Nil => ""
+          case List(Nil) => "()"
+          case _ if config.isCompletionSnippetsEnabled =>
+            val onlyParameterless = paramss.forall(_.isEmpty)
+            lazy val onlyImplicitOrTypeParams = paramss.forall(
+              _.exists { sym =>
+                sym.isType || sym.is(Implicit) || sym.is(Given)
+              }
+            )
+            if onlyParameterless then "()" * paramss.length
+            else if onlyImplicitOrTypeParams then ""
+            else if bracketSuffix == "[$0]" then "()"
+            else "($0)"
+          case _ => ""
+        end match
+      else ""
+
+    val templateSuffix =
+      if shouldAddSnippet && isNewPosition
+        && (symbol.toString.startsWith(
+          "trait"
+        ) // trait A{ def doSomething: Int}
```

Comment:
```suggestion
        && (symbol.is(Trait)) // trait A{ def doSomething: Int}
```

can you explain which test will fail? I think it shouldn't be flaky (if it is, it's a compiler bug and we should report it)

Reply:
(sorry for duplicated comment, but I want to discuss this in another thread)

Reply:
oh, I would add the test case. Generally the flags are flaky. I have it on my todo list to put the issue on dotty and add the link in here. As an example, checking the `is(Trait)` flag on some cases was retuning `false`, while calling it on `.info.typeSymbol.` was returning `true`. We discussed it with Vadim on Slack, but certainly it has to get formally documented somewhere.

Reply:
this was the type5 test that used to fail before but now passes. So I reverted to using flags 👍

Reply:
Update: the reason the demo was failing was the use of `Flags.Trait` instead of `toString.startsWith("trait ")` 
Now that I reverted to using toString, the demo works correctly. I am now scratching my head.

Reply:
Maybe it's not a trait, but a Java interface or an abstract class?

Reply:
I was thinking about that. But it is actually coming from `scala.collection.mutable`

Reply:
I wonder how I can add a test that fails with `Flags.Trait` and why `type5` is now passing regardless.

Reply:
It turned out that it was Intellisense that was messing up with things.

Reply:
I created this Issue in Dotty to explain the problem https://github.com/lampepfl/dotty/issues/15790

Reply:
Thank you for taking a look!
Let's go with the current implementation, and we can refactor later 👍